core/state/statedb.go

```diff
@@ -187,8 +187,14 @@ func (s *StateDB) setError(err error) {
 	}
 }
 
+// Error returns all memorized database failure wrapped in a single
+// error object.
 func (s *StateDB) Error() error {
-	return s.dbErr
+	errs := []error{s.dbErr}
+	for _, obj := range s.stateObjects {
+		errs = append(errs, obj.dbErr)
+	}
+	return errors.Join(errs...) // nil error will be ignored
```

Comment:
I'm not sure about this. 

- It makes the `Error()` kind of expensive, iterating through all `stateObjects`. For the bad block-case, that's fine, but this is called from other locations too
    - `txpool.validateTx` for each tx
    - `txpool.GetNonce` , on every call

Also, it makes the returned error pretty much unbounded: it can be comprised of any number of errors. Isn't it sufficient to use only the first one?

Reply:
I also think it's weird that we have to check all stateObjects for error. When an error happens in a state object, we should just assign it on the StateDB directly.

Reply:
gotcha, I will update it.